src/macaron/slsa_analyzer/analyzer.py

```diff
@@ -668,34 +668,33 @@ def to_analysis_target(
                 # If a PURL but no repository path is provided, we try to extract the repository path from the PURL.
                 # Note that we can't always extract the repository path from any provided PURL.
                 converted_repo_path = None
-                repo: str = ""
-                digest: str = ""
+                repo: str | None = None
+                digest: str | None = None
                 # parsed_purl cannot be None here, but mypy cannot detect that without some extra help.
                 if parsed_purl is not None:
                     if provenance_payload:
                         # Try to find repository and commit via provenance.
                         try:
                             repo, digest = extract_repo_and_commit_from_provenance(provenance_payload)
                         except ProvenanceError as error:
-                            logger.debug("Failed to extract repo and commit from provenance: %s", error)
-
-                    if repo and digest:
-                        return Analyzer.AnalysisTarget(
-                            parsed_purl=parsed_purl,
-                            repo_path=repo,
-                            branch="",
-                            digest=digest,
-                        )
+                            logger.debug("Failed to extract repo or commit from provenance: %s", error)
+
+                            return Analyzer.AnalysisTarget(
+                                parsed_purl=parsed_purl,
+                                repo_path=repo or "",
+                                branch="",
+                                digest=digest or "",
+                            )
```

Comment:
This return would only run when a `ProvenanceError` is raised. Is this intended? :thinking:

Reply:
It was not intended.